src/webview/tests.py

```diff
@@ -36,6 +36,267 @@
+            },
+        )
+        self.assertEqual(response.status_code, 200)
+
+        # Verify derivation2 has been removed
+        self.suggestion.refresh_from_db()
+        remaining_derivation_ids = set(
+            self.suggestion.derivations.values_list("id", flat=True)
+        )
+        self.assertEqual(len(remaining_derivation_ids), 1)
+        self.assertIn(self.derivation1.id, remaining_derivation_ids)
+        self.assertNotIn(self.derivation2.id, remaining_derivation_ids)
+
+        # Verify package2 has been removed from the cached payload
+        updated_cached_packages = self.suggestion.cached.payload["packages"]
+        self.assertIn("package1", updated_cached_packages)
+        self.assertNotIn("package2", updated_cached_packages)
+
+    def test_remove_package_from_pending_suggestion(self) -> None:
+        """Test removing a package from a suggestion in pending status (editable)"""
+
+        # Make request to keep only derivation1 (remove derivation2)
+        url = reverse("webview:suggestions_view")
+        response = self.client.post(
+            url,
+            {
+                "suggestion_id": self.suggestion.pk,
+                "derivation_ids": [str(self.derivation1.id)],
+            },
+        )
+        self.assertEqual(response.status_code, 200)
+
+        # Verify derivation2 has been removed from the suggestion
+        self.suggestion.refresh_from_db()
+        remaining_derivation_ids = set(
+            self.suggestion.derivations.values_list("id", flat=True)
+        )
+        self.assertIn(self.derivation1.id, remaining_derivation_ids)
+        self.assertNotIn(self.derivation2.id, remaining_derivation_ids)
+
+        # Verify package2 has been removed from the cached payload
+        updated_cached_packages = self.suggestion.cached.payload["packages"]
+        self.assertIn("package1", updated_cached_packages)
+        self.assertNotIn("package2", updated_cached_packages)
+
+    def test_restore_package(self) -> None:
+        """Test removing a package from a suggestion in pending status (editable)"""
+        # Request to keep only derivation1 (remove derivation2)
+        url = reverse("webview:suggestions_view")
+        response = self.client.post(
+            url,
+            {
+                "suggestion_id": self.suggestion.pk,
+                "derivation_ids": [str(self.derivation1.id)],
+            },
+        )
+        self.assertEqual(response.status_code, 200)
+
+        # Verify derivation2 has been removed from the suggestion
+        self.suggestion.refresh_from_db()
+        remaining_derivation_ids = set(
+            self.suggestion.derivations.values_list("id", flat=True)
+        )
+        self.assertIn(self.derivation1.id, remaining_derivation_ids)
+        self.assertNotIn(self.derivation2.id, remaining_derivation_ids)
+
+        # Restore package2 by including both derivation IDs again
+        restore_response = self.client.post(
```

Comment:
Can we maybe here somewhat tidily obtain the action URL from the response, to make sure it's actually made available in the UI? One way that avoids parsing the entire response HTML would be executing the template tag on the response context